### PR TITLE
[container app] Fix #28226: `az containerapp job update`: Update existing scale rules if `--scale-rule-name` is passed

### LINKED ISSUE #28226: az containerapp job update overwrites all scale rules instead of updating one
### Describe the bug

Run a command like this one:

```bash
az containerapp job update --name $NAME --resource-group $RG --scale-rule-name $SR_NAME --scale-rule-type "azure-queue" --scale-rule-metadata "accountName=$ACCOUNT_NAME" "queueName=$QUEUE_NAME" "queueLength=1" --scale-rule-auth "connection=connection-string-secret"
```

It will effectively create the desired scale rule. However, it will also delete _all other existing scale rules_.

I want to keep two scale rules.

### Related command

`az containerapp job update --name $NAME --resource-group $RG --scale-rule-name $SR_NAME --scale-rule-type "azure-queue" --scale-rule-metadata "accountName=$ACCOUNT_NAME" "queueName=$QUEUE_NAME" "queueLength=1" --scale-rule-auth "connection=connection-string-secret"`

### Errors

Other scale rules deleted

### Issue script & Debug output

N/A

### Expected behavior

The scale rule name I set is overwritten, other scale rules are not touched.

### Environment Summary

azure-cli                         2.56.0

core                              2.56.0
telemetry                          1.1.0

Extensions:
containerapp                      0.3.46

Dependencies:
msal                            1.24.0b2
azure-mgmt-resource             23.1.0b2

Python location 'C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Users\USERNAME\.azure\cliextensions'

Python (Windows) 3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:21:31) [MSC v.1936 32 bit (Intel)]

Legal docs and information: aka.ms/AzureCliLegal

### Additional context

My objective is two spawn my job if any of the queues have messages to process.

### PR DESCRIPTION
**Pull request by anandanthony**
**Related command**
`az containerapp job update`

**Description**
Fix for behavior - https://github.com/Azure/azure-cli/issues/28226

**Testing Guide**

**History Notes**

[Component Name 1] BREAKING CHANGE: `az command a`: Make some customer-facing breaking change
[Component Name 2] `az command b`: Add some customer-facing feature

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

File: src/azure-cli/azure/cli/command_modules/containerapp/custom.py
Comment (Greedygre) [inline src/azure-cli/azure/cli/command_modules/containerapp/custom.py:1226]: Recommend using `safe_get` to avoid `NoneType` error.

Comment (anandanthony) [reply src/azure-cli/azure/cli/command_modules/containerapp/custom.py]: @Greedygre, I have updated this to use `safe_get`

**Review by vinisoto — APPROVED** [review]

**Review by Greedygre — APPROVED** [review]

**Comment by Greedygre** [comment]
Hi @zhoxing-ms 

Can you help to review this PR? Thanks.

File: src/azure-cli/azure/cli/command_modules/containerapp/custom.py
Comment (Greedygre) [inline src/azure-cli/azure/cli/command_modules/containerapp/custom.py:1225]: Wait, `scale_def` is miss, please check logic in line 1204-1209

image

**Comment by Greedygre** [comment]
Hi @anandanthony 
Please rerun all tests for `az containerapp job update` to avoid regression.

**Comment by Greedygre** [comment]
Please don't merge this PR until I approve again.

File: src/azure-cli/azure/cli/command_modules/containerapp/custom.py
Comment (anandanthony) [reply src/azure-cli/azure/cli/command_modules/containerapp/custom.py]: @Greedygre I have updated the code to avoid assigning `scale_def` again with an empty model. That will not be required.

**Comment by anandanthony** [comment]
> Hi @anandanthony Please rerun all tests for `az containerapp job update` to avoid regression.

In the test - `test_containerappjob_event_triggered_crud`, I have covered all scenarios, wherein we update current scale rule, add scale rule and validate scale properties.

**Review by Greedygre — APPROVED** [review]

**Review by zhoxing-ms — APPROVED** [review]

**Pull request merged**
